pandas/core/internals/blocks.py

```diff
@@ -665,12 +714,18 @@ def astype(
             - ``ignore`` : suppress exceptions. On error return original object
         using_cow: bool, default False
             Signaling if copy on write copy logic is used.
+        squeeze : bool, default False
+            squeeze values to ndim=1 if only one column is given
 
         Returns
         -------
         Block
         """
         values = self.values
+        if squeeze and values.ndim == 2:
+            if values.shape[0] != 1:
+                raise ValueError("Can not squeeze with more than one column.")
```

Comment:
Is this just a defensive exception i.e. we should never hit this?

Reply:
yes, we shouldn't get there